composer/models/huggingface.py

```diff
@@ -403,6 +421,40 @@ def get_metadata(self):
                     }
         return {'model': model_output, 'tokenizer': tokenizer_output}
 
+    def generate(self,
+                 input_ids: torch.Tensor,
+                 num_beams: int = 1,
+                 do_sample: bool = False,
+                 max_new_tokens: int = 20,
+                 **kwargs):
+        """Generate from the underlying HuggingFace model.
+
+        Except for ``pad_token_id``, which is optionally read from ``self.tokenizer``, all args are passed along
+        to :meth:`transformers.GenerationMixin.generate` function.
+
+        Args:
+            input_ids (torch.Tensor): Input ids to generate from.
+            num_beams (int, optional): Number of beams to use for beam search. Defaults to 1.
+            do_sample (bool, optional): Whether to sample during generation. Defaults to False.
+            max_new_tokens (int, optional): Maximum number of new tokens to generate. Defaults to 20.
+            **kwargs: Additional arguments passed to :meth:`transformers.GenerationMixin.generate` function.
+        """
+        # We need to call forward once in order for FSDP + generate to work
+        # See https://github.com/huggingface/accelerate/issues/570, https://github.com/huggingface/accelerate/issues/947,
+        # and https://github.com/pytorch/pytorch/issues/82461 for more info
+        if not self.dummy_forward_called:
+            with torch.no_grad():
+                self.model(input_ids=torch.tensor([[0]], dtype=torch.long, device=input_ids.device))
+            self.dummy_forward_called = True
```

Comment:
"We have no plans to support this at the moment"  

F

I would recommend emitting a waring saying we are doing a dummy implementation, this will add some latency.

Reply:
The latency is _extremely_ minor. This is one call to forward (and only once), on one token of input. Whereas generate is num_tokens calls to forward. Maybe worth adding a warning anyway. Also this reminded me that I need to check if this works for encoder decoders too.

Reply:
fixed for t5. skipping the warning because this is so minor compared to the latency of generate.